client/src/language/diagnosticsSupport.ts

```diff
@@ -137,22 +133,39 @@ const checkIsIgnoredPylanceUndefinedVariable = async (
 
 const checkIsIgnoredShellcheckSc2154 = async (
   diagnostic: vscode.Diagnostic,
-  variablevalues: Array<{ name: string, value: string }> | undefined
+  originalTextDocument: vscode.TextDocument,
+  adjustedRange: vscode.Range
 ): Promise<boolean> => {
   if (diagnostic.source?.includes('shellcheck') !== true && diagnostic.code !== 'SC2154') {
     return false
   }
-  const message = diagnostic.message
-  const match = message.match(/^(?<variableName>\w+) is referenced but not assigned\.$/)
-  const variableName = match?.groups?.variableName
-  if (variableName === undefined) {
-    return false
-  }
 
-  if (variablevalues === undefined) {
-    // We use a static list of common directories as fallback when the scan is not done
-    return commonDirectoriesVariables.has(variableName)
+  const position = (() => {
```

Comment:
Same comment as for Pylance, do we want to keep commit "Use definition to filter out shellcheck's SC2154 ", or just keep ignoring this error all the time?